Frameworks/AudioToolbox/AudioConverter.mm

```diff
@@ -16,28 +17,186 @@
+    }
+
+    tempMediaType->SetUINT32(MF_MT_AUDIO_NUM_CHANNELS, format->mChannelsPerFrame);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_SAMPLES_PER_SECOND, format->mSampleRate);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_BLOCK_ALIGNMENT, format->mBytesPerFrame);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_AVG_BYTES_PER_SECOND, format->mSampleRate * format->mBytesPerFrame);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_BITS_PER_SAMPLE, format->mBitsPerChannel);
+    tempMediaType->SetUINT32(MF_MT_ALL_SAMPLES_INDEPENDENT, TRUE);
+
+    *mediaType = tempMediaType;
 }
 
 /**
- @Status Stub
+@Status Caveat: AudioConverter supported only for conversions through AudioConverterConvertBuffer.
+                Only Linear PCM formats are supported with 1 Frame per Packet and equal source & destination Channels per Frame.
+                Returns kAudioConverterErr_UnspecifiedError on failure and noErr (0) on success. Other return types not supported.
 */
 OSStatus AudioConverterNew(const AudioStreamBasicDescription* inSourceFormat,
                            const AudioStreamBasicDescription* inDestinationFormat,
-                           DWORD* outAudioConverter) {
-    // TODO: outAudioConverter ought to be made AudioConverterRef _Nullable * when this is unstubbed
-    UNIMPLEMENTED();
-    if (outAudioConverter) {
-        *outAudioConverter = 1234;
+                           AudioConverterRef _Nullable* outAudioConverter) {
+    if (inSourceFormat->mFormatID != kAudioFormatLinearPCM || inDestinationFormat->mFormatID != kAudioFormatLinearPCM) {
+        UNIMPLEMENTED();
+        return StubReturn();
     }
-    return StubReturn();
+
+    if (inSourceFormat->mChannelsPerFrame != inDestinationFormat->mChannelsPerFrame) {
+        UNIMPLEMENTED();
+        return StubReturn();
+    }
+
+    if (inSourceFormat->mFramesPerPacket != 1 || inSourceFormat->mFramesPerPacket != inDestinationFormat->mFramesPerPacket) {
+        UNIMPLEMENTED();
+        return StubReturn();
+    }
+
+    HRESULT status;
+
+    status = CoInitializeEx(nullptr, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"COM Initialization Failed");
+        return kAudioConverterErr_UnspecifiedError;
+    }
+
+    status = MFStartup(MF_VERSION, MFSTARTUP_NOSOCKET);
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"Media Foundation Initialization Failed");
+        return kAudioConverterErr_UnspecifiedError;
+    }
+
+    IUnknown* spTransformUnk;
+
+    status = CoCreateInstance(CLSID_AudioResamplerMediaObject, nullptr, CLSCTX_INPROC_SERVER, IID_IUnknown, (void**)&spTransformUnk);
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"Audio Resampler Initialization Failed");
+        spTransformUnk->Release();
+        return kAudioConverterErr_UnspecifiedError;
+    }
+
+    AudioConverterRef outConverter = new OpaqueAudioConverter;
+    outConverter->transform = nullptr;
```

Comment:
this seems awfully reminiscent of a constructor that would encapsulate what you want better. ALong with other comments about the lifetime and ownership of this object, please consider an alternate approach.